config/application.php

```diff
@@ -121,10 +110,59 @@
     $_SERVER['HTTPS'] = 'on';
 }
 
-$env_config = __DIR__ . '/environments/' . WP_ENV . '.php';
+// FS writes aren't permitted in test or live, so we should let WordPress know to
+// disable relevant UI.
+if (in_array($_ENV['PANTHEON_ENVIRONMENT'], array('test', 'live'))) {
+    if (!env('DISALLOW_FILE_MODS')) {
+        Config::define('DISALLOW_FILE_MODS', true);
+    }
+    if (!env('DISALLOW_FILE_EDIT')) {
+        Config::define('DISALLOW_FILE_EDIT', true);
+    }
+}
+
+/**
+ * Set WP_ENVIRONMENT_TYPE according to the Pantheon Environment
+ */
+if (getenv('WP_ENVIRONMENT_TYPE') === false) {
+    switch ($_ENV['PANTHEON_ENVIRONMENT']) {
+        case 'live':
+            putenv('WP_ENVIRONMENT_TYPE=production');
+            break;
+        case 'test':
+            putenv('WP_ENVIRONMENT_TYPE=staging');
+            break;
+        default:
+            putenv('WP_ENVIRONMENT_TYPE=development');
+            break;
+    }
+}
+
+/**
+ * For developers: WordPress debugging mode.
+ *
+ * Change this to true to enable the display of notices during development.
+ * It is strongly recommended that plugin and theme developers use WP_DEBUG
+ * in their development environments.
+ *
+ * You may want to examine $_ENV['PANTHEON_ENVIRONMENT'] to set this to be
+ * "true" in dev, but false in test and live.
+ */
+if (!env('WP_DEBUG')) {
```

Comment:
I'm assuming the `env()` function works like this: https://gist.github.com/mattstauffer/59170bc2cb6318ab6d46

In that case, if there was no WP_DEBUG value, it would also return false, making the conditional true, and disabling WP_DEBUG. While this is not necessarily a problem, it would execute line 152 if

* no value was set for WP_DEBUG
* a value was set for WP_DEBUG and it was `false`

Since WP_DEBUG _defaults_ to `false`, perhaps this would be better if the conditional was flipped, e.g.

```php
if ( env('WP_DEBUG') === true ) {
    Config::define( 'WP_DEBUG', true );
}
```